src/main/java/net/dv8tion/jda/core/managers/GuildController.java

```diff
@@ -2117,6 +2117,10 @@ public RoleAction createCopyOfRole(Role role)
         Checks.notNull(icon, "Emote icon");
         Checks.notNull(roles, "Roles");
 
+        // Check for emote upload limits
+        Checks.check(getGuild().getEmoteCache().stream().filter(emote -> !emote.isAnimated()).count() < 50, "Guild normal emote limit has already been reached!");
+        Checks.check(getGuild().getEmoteCache().stream().filter(Emote::isAnimated).count() < 50, "Guild animated emote limit has already been reached!");
```

Comment:
If one bucket is full we just don't allow adding to the other?

Reply:
whoops, wasn't thinking there

Reply:
Actually now that I am looking at it, I am not quite sure how to do this.
We have no way of telling whether or not the `Icon` is animated before the request.

Reply:
Technically we can, discord does check it as well. Maybe this is too much to check, just document the error responses then.